modules/terceptAnalyticsAdapter.js

```diff
@@ -0,0 +1,145 @@
+      } else if (eventType === CONSTANTS.EVENTS.BID_RESPONSE) {
+        mapBidResponse(args, 'response');
+      } else if (eventType === CONSTANTS.EVENTS.BID_WON) {
+        send({
+          bidWon: mapBidResponse(args, 'win')
+        }, 'won');
+      }
+    }
+
+    if (eventType === CONSTANTS.EVENTS.AUCTION_END) {
+      send(events, 'auctionEnd');
+    }
+  }
+});
+
+function mapBidRequests(params) {
+  let arr = [];
+  if (typeof params.bids !== 'undefined' && params.bids.length) {
+    params.bids.forEach(function (bid) {
+      arr.push({
+        bidderCode: bid.bidder,
+        bidId: bid.bidId,
+        adUnitCode: bid.adUnitCode,
+        requestId: bid.bidderRequestId,
+        auctionId: bid.auctionId,
+        transactionId: bid.transactionId,
+        sizes: utils.parseSizesInput(bid.mediaTypes.banner.sizes).toString(),
+        renderStatus: 1,
+        requestTimestamp: params.auctionStart
+      });
+    });
+  }
+  return arr;
+}
+
+function mapBidResponse(bidResponse, status) {
+  if (status !== 'win') {
+    let bid = events.bids.filter(o => o.bidId == bidResponse.bidId || o.bidId == bidResponse.requestId)[0];
+    Object.assign(bid, {
+      bidderCode: bidResponse.bidder,
+      bidId: status == 'timeout' ? bidResponse.bidId : bidResponse.requestId,
+      adUnitCode: bidResponse.adUnitCode,
+      auctionId: bidResponse.auctionId,
+      creativeId: bidResponse.creativeId,
+      transactionId: bidResponse.transactionId,
+      currency: bidResponse.currency,
+      cpm: bidResponse.cpm,
+      netRevenue: bidResponse.netRevenue,
+      mediaType: bidResponse.mediaType,
+      statusMessage: bidResponse.statusMessage,
+      status: bidResponse.status,
+      renderStatus: status == 'timeout' ? 3 : 2,
+      timeToRespond: bidResponse.timeToRespond,
+      requestTimestamp: bidResponse.requestTimestamp,
+      responseTimestamp: bidResponse.responseTimestamp
+    });
+  } else if (status == 'win') {
+    return {
+      bidderCode: bidResponse.bidder,
+      bidId: bidResponse.requestId,
+      adUnitCode: bidResponse.adUnitCode,
+      auctionId: bidResponse.auctionId,
+      creativeId: bidResponse.creativeId,
+      transactionId: bidResponse.transactionId,
+      currency: bidResponse.currency,
+      cpm: bidResponse.cpm,
+      netRevenue: bidResponse.netRevenue,
+      renderedSize: bidResponse.size,
+      mediaType: bidResponse.mediaType,
+      statusMessage: bidResponse.statusMessage,
+      status: bidResponse.status,
+      renderStatus: 4,
+      timeToRespond: bidResponse.timeToRespond,
+      requestTimestamp: bidResponse.requestTimestamp,
+      responseTimestamp: bidResponse.responseTimestamp
+    }
+  }
+}
+
+function send(data, status) {
+  let location = utils.getWindowLocation();
+  if (typeof data !== 'undefined' && typeof data.auctionInit !== 'undefined') {
+    data.auctionInit = Object.assign({ host: location.host, path: location.pathname, search: location.search }, data.auctionInit);
```

Comment:
Could `data.auctionInit` already have host, path, and search properties?

If not, this can be replaced with:
```
Object.assign(data.auctionInit, { host: location.host, path: location.pathname, search: location.search });
```

Reply:
I have replaced the line of code with your recommended change.